### ISSUE TITLE
CPU running time of pretrained model  RetinaFace-R50

### ISSUE THREAD

**Issue by marcobarnobi** [deepinsight/insightface#1096]
I am running the pre-trained model RetinaFace-R50 and with my mac pro it takes about 4s (0.25fps) to process an image (only inference time, w/o loading/pre/post-processing).

The paper says that the bigger RetinaFace model, with ResNet152 backbone, runs in. 75 ms (13fps). This is more than 50 times faster than what I experience with a CPU and on a smaller model. Is this normal? 

To draw an analogy, using the GPU on the MobileNet-0.25 model, the execution is about 12 times faster than with the CPU. However in my case I am using the CPU on a smaller model and, despite that, it is 50 times slower. 

From the paper: 
<img width="346" alt="Screenshot 2020-04-17 at 10 59 30" src="https://user-images.githubusercontent.com/17983807/79564011-287a7400-80ae-11ea-8212-1e08c7475858.png">

**Comment by bartolootrit** [comment]
The inference time also depends on the resolution/quality of the input. It may take 1 second to process 658x480 video frame and 4 seconds to process 1280x712 video frame on an ordinary CPU. A reduction of input resolution/quality during the preprocessing phase may help.

**Comment by marcobarnobi** [comment]
> It may take 1 second to process 658x480 video frame and 4 seconds to process 1280x712 video frame on an ordinary CPU. A reduction of input resolution/quality during the preprocessing phase may help. If the target faces are big enough, it shouldn't have any practical impact on detection accuracy.

It makes sense that the input size (to which all input images are resized) matters. However I see that the model performance changes as the input size is modified.

I satisfied my need for speed using a model with a lighter backbone, which still performs pretty well. This speeds up the computation of a x5 factor on my CPU. 
https://github.com/deepinsight/insightface/issues/669#issue-442669691

:)
